numpyro/infer/elbo.py

```diff
@@ -728,43 +730,72 @@ def get_importance_trace_enum(model, guide, args, kwargs, params, max_plate_nest
             guide_trace = _trace(guide).get_trace(*args, **kwargs)
         model = substitute(replay(model, guide_trace), data=params)
         model_trace = _trace(model).get_trace(*args, **kwargs)
-    for is_model, tr in zip((False, True), (guide_trace, model_trace)):
+
+    sum_vars = frozenset()
+    for is_model, tr in zip((True, False), (model_trace, guide_trace)):
         for name, site in tr.items():
             if site["type"] == "sample":
-                if is_model and (site["is_observed"] or (site["name"] in guide_trace)):
-                    site["is_measure"] = False
-                if "log_prob" not in site:
-                    value = site["value"]
-                    intermediates = site["intermediates"]
+                value = site["value"]
+                intermediates = site["intermediates"]
+                dim_to_name = site["infer"]["dim_to_name"]
+
+                # compute log factor
+                if (
+                    is_model
+                    and (name in guide_trace)
+                    and model_deps[name].isdisjoint(sum_vars | guide_desc[name])
```

Comment:
@martinjankowiak can you check this? I think `model_deps[name].isdisjoint(guide_desc[name])` should take care of the condition for analytic kl now.

Reply:
can you remind me of some of the context? does this PR support both model-side and guide-side enumeration? what do i need to know about what's happening on the discrete side?

Reply:
Yes, it supports both model-side and guide-side enumeration. It also uses provenance tracking to get the dependency structure of terms in the model on model-enumerated sites. Then this info is used to partition model terms into uncontracted terms and terms that are being contracted together. Then each cost term from model/guide is integrated wrt guide-enumerated sites.

Reply:
do we have any tests that cover the KL branch specifically? more generally what tests do we have to cover the case with continuous latents + TraceEnum?

suppose the model is a K-component gaussian mixture. for model side-enumeration we can't compute the KL because we don't have the formula. for guide-side enumeration with a K-component gaussian mixture guide, we can compute both (i.e. one continuous and one discrete) KL divergences, right?

can we check these two simple cases somehow?

is it the case that model-side enumerations makes most KL divergence computations intractable?

Reply:
I will try to write some tests.